src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -839,6 +839,20 @@ public INamedTypeSymbol CreateTupleTypeSymbol(ImmutableArray<ITypeSymbol> elemen
 
         protected abstract INamedTypeSymbol CommonCreateTupleTypeSymbol(ImmutableArray<ITypeSymbol> elementTypes, ImmutableArray<string> elementNames);
 
+        /// <summary>
+        /// Returns a new INamedTypeSymbol with the given underlying type and (optional) element names.
+        /// </summary>
+        /// <remarks>
+        /// Since VB doesn't support tuples yet, this call will fail in a VB compilation.
+        /// Also, the underlying type needs to be tuple-compatible.
+        /// </remarks>
+        public INamedTypeSymbol CreateTupleTypeSymbol(INamedTypeSymbol underlyingType, ImmutableArray<string> elementNames = default(ImmutableArray<string>))
```

Comment:
I'm afraid your answers didn't help me understand. We'll have to discuss.

I want to note one additional scenario to consider (metadata scenario). I have a 9-tuple with metadata for the 9 element names and the nested type happens to be a 2-tuple with names as well.